### PR TITLE
Bot should only run if the number of holes detected is valid

### PR DESCRIPTION
**Pull request by brandonabela**

### CODE REVIEW

File: Testing/offline_testing.py [hunk 1]
```diff
@@ -112,48 +112,49 @@ def optimal_path_video(self, video_path, show_video, save_video):
...
-
-                for a_target in target_holes:
-                    cv2.circle(modified_frame, (a_target[0], a_target[1]), 2, (0, 0, 0), 3)
-                
-                shrink_border = ball_path.get_shrink_borders()
-                
-                for i, _ in enumerate(shrink_border):
-                    if i % 2 != 0:
-                        cv2.line(modified_frame, shrink_border[i], shrink_border[(i + 1) % len(shrink_border)], (150, 150, 255), 3)
+                if bot.holes:
```
Comment (KeithGeorgeCiantar) [inline Testing/offline_testing.py:115]: Does this check the number of holes or does it only check whether any list of holes exist?

File: Logic/bot.py [hunk 2]
```diff
@@ -25,12 +25,15 @@ def find_holes(self, frame):
         '''Responsible for finding the holes if not set'''
 
         if not self.holes:
-            self.holes = self.ball_detection.find_holes(frame)
+            corner_holes = self.ball_detection.find_corner_holes(frame)
 
-            board_positions = self.ball_detection.board_boundary(self.holes)
+            if len(corner_holes) == 4:
```
Comment (KeithGeorgeCiantar) [inline Logic/bot.py:30]: How does this know that the hole position is correct? Or does the corner hole detection have strict rules and only finds the 4 corners?

**Review by KeithGeorgeCiantar — COMMENTED** [review]
Some small questions, otherwise seems to be ok.

File: Testing/offline_testing.py (hunk 1 above)
Comment (brandonabela) [reply Testing/offline_testing.py:115]: This only checks that a list of holes is detected.

File: Logic/bot.py (hunk 2 above)
Comment (brandonabela) [reply Logic/bot.py:30]: This does not ensure that the 4 detected holes are correct, however, when applying the HoughCircles it uses the minimum and maximum radius of the corner holes.

**Review by KeithGeorgeCiantar — APPROVED** [review]
As long as everything works, these changes look good to me.

**Pull request merged**